pkgs/applications/window-managers/river/default.nix

```diff
@@ -13,10 +13,13 @@
 , udev
 , libevdev
 , libinput
-, libX11
 , libGL
+, xwaylandSupport ? true, libX11 ? null
 }:
 
+assert xwaylandSupport -> (libX11 != null);
+
+with lib;
```

Comment:
Don't introduce `lib` in the entire scope of `stdenv.mkDerivation`, leave it as it was before.